src/ui/pages/BridgeForm/BridgeLine/BridgeLine.tsx

```diff
@@ -0,0 +1,177 @@
+import { useAssetsPrices, type Asset } from 'defi-sdk';
+import React, { useMemo } from 'react';
+import { useCurrency } from 'src/modules/currency/useCurrency';
+import type { Chain } from 'src/modules/networks/Chain';
+import { getCommonQuantity } from 'src/modules/networks/asset';
+import type { Quote } from 'src/shared/types/Quote';
+import { formatSeconds } from 'src/shared/units/formatSeconds';
+import { SlidingRectangle } from 'src/ui/components/SlidingRectangle/SlidingRectangle';
+import type { QuotesData } from 'src/ui/shared/requests/useQuotes';
+import { noValueDash } from 'src/ui/shared/typography';
+import { HStack } from 'src/ui/ui-kit/HStack';
+import { UIText } from 'src/ui/ui-kit/UIText';
+import { TokenIcon } from 'src/ui/ui-kit/TokenIcon';
+import { formatCurrencyValue } from 'src/shared/units/formatCurrencyValue';
+import { getQuotesErrorMessage } from '../../SwapForm/Quotes/getQuotesErrorMessage';
+
+function getFeePriceValue({
+  quote,
+  chain,
+  asset,
+}: {
+  quote: Quote;
+  chain: Chain;
+  asset: Asset;
+}) {
+  if (asset.price?.value == null) {
+    return null;
+  }
+
+  const quantity = getCommonQuantity({
+    baseQuantity: quote.bridge_fee_amount,
+    chain,
+    asset,
+  });
+
+  return quantity.times(asset.price.value);
+}
+
+export function BridgeLine({
+  spendChain,
+  quotesData,
+  quote,
+}: {
+  spendChain: Chain;
+  quotesData: QuotesData;
+  quote: Quote | null;
+}) {
+  const { isLoading, error } = quotesData;
+
+  const { currency } = useCurrency();
+  const { value: feeAssetValue } = useAssetsPrices({
+    asset_codes: [quote?.bridge_fee_asset_id || ''],
+    currency,
+  });
+
+  const feeAssetId = quote?.bridge_fee_asset_id;
+  const feeAsset = feeAssetId ? feeAssetValue?.[feeAssetId] : undefined;
+  const feePriceValue = useMemo(
+    () =>
+      quote && feeAsset
+        ? getFeePriceValue({
+            quote,
+            chain: spendChain,
+            asset: feeAsset,
+          })
+        : null,
+    [feeAsset, quote, spendChain]
+  );
+
+  return (
+    <HStack
+      gap={12}
+      justifyContent="space-between"
+      alignItems="center"
+      style={{
+        visibility: !isLoading && !quote && !error ? 'hidden' : undefined,
+      }}
+    >
+      <UIText kind="small/regular" color="var(--neutral-700)">
+        Bridge
+      </UIText>
+      <span>
+        {isLoading && !quote ? (
```

Comment:
It looks like we allow user to see and use quote when the loading is not `done` yet?
May be we need to add such check here and to the main form?

Reply:
Isn't it the point to show the progression of quotes? But we don't allow submit during `isLoading`